### PR TITLE
RavenDB-12720: Create configuration for logs retention time

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Server/Config/Categories/LogsConfiguration.cs [hunk 1]
```diff
@@ -25,5 +25,12 @@ public class LogsConfiguration : ConfigurationCategory
         [SizeUnit(SizeUnit.Megabytes)]
         [ConfigurationEntry("Logs.MaxFileSizeInMb", ConfigurationEntryScope.ServerWideOnly)]
         public Size MaxFileSize { get; set; }
+
+        [Description("How far back we should retain log entries")]
+        [DefaultValue(3 * 24)]
+        [TimeUnit(TimeUnit.Hours)]
+        [ConfigurationEntry("Logs.RetentionTimeInHrs", ConfigurationEntryScope.ServerWideOnly)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/LogsConfiguration.cs:32]: Given that this is new configuration, why do we have `Hrs` here? Do we usually have both `Hrs` and `Hours`?

Comment (garayx) [reply src/Raven.Server/Config/Categories/LogsConfiguration.cs:32]: We are using _shortcuts_ (ms, sec, min, hrs) on TimeSetting fields in most configurations. Should I stick to it? or use _Hours_ ?

Comment (ayende) [reply src/Raven.Server/Config/Categories/LogsConfiguration.cs:32]: @ppekrol What should be our default position here?
Do we want to have `Hrs` and `Hours`, just one?

Comment (ppekrol) [reply src/Raven.Server/Config/Categories/LogsConfiguration.cs:32]: `Hrs` only

File: src/Raven.Client/ServerWide/Operations/Logs/GetLogsConfigurationOperation.cs [hunk 2]
```diff
@@ -56,5 +57,10 @@ public class GetLogsConfigurationResult
         /// Indicates if logs will be written in UTC or in server local time
         /// </summary>
         public bool UseUtcTime { get; set; }
+
+        /// <summary>
+        /// Logs retention time
+        /// </summary>
+        public TimeSpan LogRetention { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/Logs/GetLogsConfigurationOperation.cs]: rename to `RetentionTime`

File: src/Raven.Client/ServerWide/Operations/Logs/SetLogsConfigurationOperation.cs [hunk 3]
```diff
@@ -16,6 +16,7 @@ public class SetLogsConfigurationOperation : IServerOperation
         public class Parameters
         {
             public LogMode Mode { get; set; }
+            public TimeSpan LogRetention { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/Logs/SetLogsConfigurationOperation.cs]: rename to `RetentionTime`

Comment (ppekrol) [reply src/Raven.Client/ServerWide/Operations/Logs/SetLogsConfigurationOperation.cs]: Also needs to be nullable to maintain backward compatibility

File: src/Raven.Server/Config/Categories/LogsConfiguration.cs [hunk 4]
```diff
@@ -25,5 +25,12 @@ public class LogsConfiguration : ConfigurationCategory
...
+        [ConfigurationEntry("Logs.RetentionTimeInHours", ConfigurationEntryScope.ServerWideOnly)]
+        public TimeSetting LogRetention { get; set; }
```
Comment (ppekrol) [inline src/Raven.Server/Config/Categories/LogsConfiguration.cs]: rename property to `RetentionTime`

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 5]
```diff
@@ -42,7 +43,7 @@ public async Task SetConfiguration()
 
                 var configuration = JsonDeserializationServer.Parameters.SetLogsConfigurationParameters(json);
 
-                LoggingSource.Instance.SetupLogMode(configuration.Mode, Server.Configuration.Logs.Path.FullPath);
+                LoggingSource.Instance.SetupLogMode(configuration.Mode, Server.Configuration.Logs.Path.FullPath, configuration.LogRetention);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: if this will be null, then use the value from configuration

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Config/Categories/LogsConfiguration.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Config/Categories/LogsConfiguration.cs:32]: done

File: src/Raven.Client/ServerWide/Operations/Logs/SetLogsConfigurationOperation.cs (hunk 3 above)
Comment (garayx) [reply src/Raven.Client/ServerWide/Operations/Logs/SetLogsConfigurationOperation.cs]: done

File: src/Raven.Server/Config/Categories/LogsConfiguration.cs (hunk 4 above)
Comment (garayx) [reply src/Raven.Server/Config/Categories/LogsConfiguration.cs]: done

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 5 above)
Comment (garayx) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: done

**Pull request merged**